### PR TITLE
Fix timezone conversion to UTC

### LINKED ISSUE #18: Fix failing tests
Looks like I wrote the tests that are deterministic on the _current_ timezone and asserting fields that will change. I don't think the code is wrong; I think the tests are wrong.

### PR DESCRIPTION
**Pull request by dbernheisel**
Resolves #18 

We were using the current timestamp (essentially `DateTime.utc_now()`) when converting to UTC, which uses the UTC offset of today instead of the UTC offset in the context of the parsed datetime.

In other words, `to_utc: true` would always convert using today's UTC offset instead of the parsed datetime's. So if today was Daylight Time, and the parsed datetime was in Standard Time, the parsed datetime would be converted to UTC using the daylight time UTC offset. This resulted in incorrect conversion.

### CODE REVIEW

**Comment by dbernheisel** [comment]
@tmr08c you mind double-checking me with some of these changed examples and ensuring it's converting correctly now?

File: lib/date_time.ex [hunk 1]
```diff
@@ -72,17 +72,17 @@ defmodule DateTimeParser.DateTime do
     )
   end
 
-  defp timezone_from_tokens(tokens) do
+  defp timezone_from_tokens(tokens, naive_datetime) do
     with zone <- format_token(tokens, :zone_abbr),
          offset <- format_token(tokens, :utc_offset),
          true <- Enum.any?([zone, offset]) do
-      Timex.Timezone.get(offset || zone)
+      Timex.Timezone.get(offset || zone, naive_datetime)
```
Comment (tmr08c) [inline lib/date_time.ex]: To make sure I understand the PR, is this the line that fixes things? 

Based on the commit and docs, it sounds like `Timex.Timezone.get/1` will default the second argument (the timezone) to the system's current time ("if one is not provided, then the current date and time is returned"). This means that if you are parsing a date that is in saving time but running the test on a date that is not in saving time, there would be a discrepancy (and vice versa). 

Instead, this is now passed in the timestamp being parsed, and that will be used as a part of returning timezone info (as opposed to being based on when the test is run).

File: lib/date_time_parser.ex [hunk 2]
```diff
@@ -72,12 +72,12 @@ defmodule DateTimeParser do
     {:ok, DateTime.from_naive!(~N[2018-01-01T15:24:00Z], "Etc/UTC")}
     # or ~U[2018-01-01T15:24:00Z] in Elixir 1.9.0+
 
-    iex> DateTimeParser.parse_datetime(~s|"Dec 1, 2018 7:39:53 AM PST"|, to_utc: true)
-    {:ok, DateTime.from_naive!(~N[2018-12-01T14:39:53Z], "Etc/UTC")}
+    iex> DateTimeParser.parse_datetime(~s|"Mar 28, 2018 7:39:53 AM PDT"|, to_utc: true)
```
Comment (tmr08c) [inline lib/date_time_parser.ex]: Was this change to have `PDT` and `PST`? That seems valuable. I sort of also like the idea of having two of the same dates, one with `to_utc: true` and one with `to_utc: false`, but not sure if that provides more value.

File: lib/date_time.ex (hunk 1 above)
Comment (dbernheisel) [reply lib/date_time.ex]: right!

File: lib/date_time_parser.ex (hunk 2 above)
Comment (dbernheisel) [reply lib/date_time_parser.ex]: Daylight Time is from Mar10 to Nov3, so I changed the date here to be within PDT so the doctest will pass. There's another one that's in PST.

I'll add another `to_utc: false` test to mirror this.

Comment (dbernheisel) [reply lib/date_time_parser.ex]: Daylight Time is from Mar10 to Nov3, so I changed the date here to be within PDT so the doctest will pass. There's another one that's in PST.

I'll add another to_utc: false test to mirror this.

**Comment by dbernheisel** [comment]
released in 1.0.0-rc2

**Pull request merged**